.github/actions/generate-commit-list-v1/src/getParsedCommitList.ts

```diff
@@ -0,0 +1,42 @@
+import getCommitType from './getCommitType'
+import { GetParsedCommitListParams, ParsedCommitList } from './types'
+
+/**
+ * Parses the raw commit list into an array of commit objects.
+ * @param {object} GetParsedCommitListParams Raw commit list and repository URL.
+ * @returns An array of parsed commits.
+ */
+
+export default function getParsedCommitList({
+  rawCommitList,
+  repositoryURL
+}: GetParsedCommitListParams): Array<ParsedCommitList> {
+  const parsedCommits: Array<ParsedCommitList> = []
+
+  for (const commit of rawCommitList) {
+    /**
+     * Regex to group the following:
+     * 1. Commit SHA (e.g. 4d6220e) 8 characters long
+     * 2. Commit message (e.g. "feat: add new feature")
+     * 3. Pull request ID (e.g. #1234)
+     */
+    const regex = /([a-z0-9]{1,8}) (.*) \((#[0-9]*)\)/
+    const commitMatches = commit.match(regex)
+    //eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const [, sha, title, id] = commitMatches!
+    const idParsed = id.replace('#', '')
+    const type = getCommitType(title)
+    const link = `${repositoryURL}/pull/${idParsed}`
```

Comment:
This assumes the commit has the pr number in the title (not always a guarantee). For commits without the pr number in the title you can [use the SHA to get the number using `git ls-remote`](https://stackoverflow.com/a/48085242/2124254)

```
git ls-remote origin 'pull/*/head' | grep <SHA> |  awk -F'/' '{print $3}'
```

Alternatively instead of `awk` you could use `cut` or any other methods to get the pull request number out of the string

Reply:
This is not going to work, this pulls the list of commits using the _last_ commit SHA of that pull request not the merging commit SHA.

Reply:
That's OK. I think we can try it as a fallback and if it doesn't find anything we just default to `null`. It should help fill in quite a few missing ids though